pkg/preflight/run.go

```diff
@@ -255,14 +260,50 @@ func RunPreflights(interactive bool, output string, format string, args []string
 	stopProgressCollection()
 	progressCollection.Wait()
 
+	if len(analyzeResults) == 0 {
+		return types.NewExitCodeError(constants.EXIT_CODE_CATCH_ALL, errors.New("no data has been collected"))
+	}
+
 	if interactive {
-		if len(analyzeResults) == 0 {
-			return errors.New("no data has been collected")
+		err = showInteractiveResults(preflightSpecName, output, analyzeResults)
+	}
+
+	err = showTextResults(format, preflightSpecName, output, analyzeResults)
+
+	var exitCode int
+	if err != nil {
+		// Always return a 2 here
+		exitCode = 2
+	} else {
+		exitCode = checkOutcomesToExitCode(analyzeResults)
+	}
```

Comment:
this feels like it should be a standard exit code here as it is not a spec issue

```suggestion
	if err != nil {
		return errors.Wrap(err, "failed to render preflight output")
	}

	exitCode := checkOutcomesToExitCode(analyzeResults)
```